src/Sylius/Component/Payment/Factory/PaymentRequestFactory.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Payment\Factory;
+
+use Sylius\Component\Payment\Model\PaymentInterface;
+use Sylius\Component\Payment\Model\PaymentMethodInterface;
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+/**
+ * @implements PaymentRequestFactoryInterface<PaymentRequestInterface>
+ */
+final class PaymentRequestFactory implements PaymentRequestFactoryInterface
+{
+    public function __construct(private FactoryInterface $factory)
+    {
+    }
+
+    public function createNew(): PaymentRequestInterface
+    {
+        return $this->factory->createNew();
+    }
+
+    public function createWithPaymentAndPaymentMethod(PaymentInterface $payment, PaymentMethodInterface $paymentMethod): PaymentRequestInterface
+    {
+        $paymentRequest = $this->createNew();
+
+        $paymentRequest->setPayment($payment);
```

Comment:
No it can be different.
For ex if the `$payment->isNew()` and a `PaymentMethod` is already set, then:

1. A `PaymentRequest` is running but the `Payment` status is not yet changed.
2. The customer choose to change `PaymentMethod`

The `Payment` will be the same but the `PaymentMethod` will be different.